web/libs/frontend-test/src/cypress/support/commands.ts

```diff
@@ -0,0 +1,119 @@
+import Loggable = Cypress.Loggable
+import Timeoutable = Cypress.Timeoutable
+import Tresholdable = Cypress.Tresholdable
+import CompareScreenshotOptions = Cypress.CompareScreenshotOptions
+import { addMatchImageSnapshotCommand } from 'cypress-image-snapshot/command';
+
+addMatchImageSnapshotCommand({
+  failureThreshold: 0.1,
+  failureThresholdType: 'percent',
+});
+
+const Screenshots = new Map<string, string>();
+
+const getName = (suffix: string) => {
+  const spec = Cypress.spec.name;
+
+  return `${spec.replace(/.([jt]s)/, '')}-${suffix}`.toLowerCase();
+};
+
+Cypress.Commands.add('captureScreenshot', {
+  prevSubject: ['element'],
+}, (subject, name, screenshotCaptureOptions: Partial<Loggable & Timeoutable & CompareScreenshotOptions> = {}) => {
+  const { withHidden = [], ...screenshotOptions } = screenshotCaptureOptions;
+  const log = Cypress.log({
+    $el: subject,
+    name: 'captureScreenshot',
+    displayName: 'captureScreenshot',
+    type: 'parent',
+    autoEnd: false,
+  });
+
+  const screenshotName = getName(name);
+
+  if (Screenshots.has(screenshotName)) {
+    throw new Error('Screenshot already taken. Did you forget to call `compareScreenshot`?');
+  }
+
+  const obj = cy.wrap(subject, { log: false });
+
+  obj.scrollIntoView({ log: false });
+
+  for (const hiddenSelector of withHidden) {
+    cy.get(hiddenSelector).invoke('css', 'visibility', 'hidden');
+  }
+  obj.screenshot(screenshotName + '-orig', Object.assign(
+    { log: false },
+    screenshotOptions,
+    {
+      onAfterScreenshot(_el, screenshot) {
+        screenshotOptions.onAfterScreenshot?.(_el, screenshot);
+        Screenshots.set(screenshotName, screenshot.path);
+      },
+    },
+  ));
+  for (const hiddenSelector of withHidden) {
+    cy.get(hiddenSelector).invoke('css', 'visibility', '');
+  }
+  log.end();
+  return obj;
+});
+
+Cypress.Commands.add('compareScreenshot', {
+  prevSubject: ['element'],
+}, (subject, name, compare, screenshotCompareOptions: Partial<Loggable & Timeoutable & CompareScreenshotOptions & Tresholdable> = {}) => {
+  const { treshold = 0.1, withHidden = [], ...screenshotOptions } = screenshotCompareOptions;
```

Comment:
I think we have a small type-i here `treshold`
and I see 0.1 is earlier in the file as `failureThreshold` can you define it as as a const value so it can be reused since it looks like we're using this value in more than 1 spot